ui/js/controllers/bugfiler.js

```diff
@@ -127,8 +127,13 @@ treeherder.controller('BugFilerCtrl', [
             summary = summary.split(" | ");
 
             for (var i=0; i < $scope.omittedLeads.length; i++) {
-                if (summary[0].search($scope.omittedLeads[i]) >= 0 && summary.length > 1) {
-                    summary.shift();
+                // In the case where only exact substring matches will provide usable bugs,
+                // Dropping things from omittedLeads will cause the substring to not match.
+                // Don't drop the lead in this case.
+                if ($scope.search_terms.length > 0 && !$scope.search_terms[0].includes($scope.omittedLeads[i])) {
```

Comment:
Is there always only 1 or 0 search terms?  If there's 2, you'll only be covering the first here.  I looked through quite a few and only found this one that had two: https://treeherder.mozilla.org/api/project/mozilla-inbound/jobs/112331329/bug_suggestions/.  Perhaps you need to check against both?  Or do you feel that checking only one covers it?